packages/backend/src/utils/npm.ts

```diff
@@ -120,6 +109,50 @@ class Command {
     }
   }
 
+  private async shouldBeUpdated(packageJson: any): Promise<boolean> {
+    const queryUrl = this.getSingleGenQueryURL(packageJson.name);
+    const npmJsModules = await npmFetch.json(queryUrl);
+    const npmJsModule: any = _.get(npmJsModules, "objects.[0]");
+    return npmJsModule ? npmJsModule.package.version !== packageJson.version : false;
+  }
+
+  public getGlobalNodeModulesPath(): string {
+    return this.globalNodeModulesPath;
+  }
+
+  public async getPackagesMetadata(query = "", author = "") {
+    const gensQueryUrl = NpmCommand.getGensQueryURL(query, author);
+    return await npmFetch.json(gensQueryUrl);
+  }
+
+  public async getPackageJson(packagePath: string): Promise<any> {
+    const packageJsonString: string = await readFile(path.join(packagePath, "package.json"), "utf8");
+    return JSON.parse(packageJsonString);
+  }
+
+  public async getPackageNamesWithOutdatedVersion(packagePaths: string[]): Promise<string[]> {
+    const packageJsonPromises: any[] = packagePaths.map((packageJsonPath) => this.getPackageJson(packageJsonPath));
+    const packageJsons: any[] = await Promise.all(packageJsonPromises);
```

Comment:
what you intend here in case one/any promises rejected ?

Reply:
an exception isnow handled in getPackageJson with try/catch